Frameworks/Foundation/NSKVOSwizzling.mm

```diff
@@ -253,21 +243,65 @@ static void notifyingVariadicSetImpl(id self, SEL _cmd, ...) {
 
     [self willChange:NSKeyValueChangeReplacement valuesAtIndexes:indexes forKey:key];
 
-    struct objc_super super {
-        self, ABI_SUPER(self)
-    };
+    objc_super super{ self, ABI_SUPER(self) };
     auto imp = (void (*)(id, SEL, NSIndexSet*, NSArray*))objc_msg_lookup_super(&super, _cmd);
     imp(self, _cmd, indexes, objects);
 
     [self didChange:NSKeyValueChangeReplacement valuesAtIndexes:indexes forKey:key];
 }
 
+static void NSKVONotifying$addXxx$(id self, SEL _cmd, NSSet* set) {
+    NSString* key = _keyForSelector(self, _cmd);
+
+    [self willChangeValueForKey:key withSetMutation:NSKeyValueUnionSetMutation usingObjects:set];
+
+    objc_super super{ self, ABI_SUPER(self) };
+    auto imp = (void (*)(id, SEL, NSSet*))objc_msg_lookup_super(&super, _cmd);
+    imp(self, _cmd, set);
+
+    [self didChangeValueForKey:key withSetMutation:NSKeyValueUnionSetMutation usingObjects:set];
+}
+
+static void NSKVONotifying$removeXxx$(id self, SEL _cmd, NSSet* set) {
```

Comment:
These are all the same with one parameter variant, make this into a utility function and pass the variant parameter.